### ISSUE TITLE
[Feature] Check input for pipe

### ISSUE THREAD

**Issue by Beanow** [plaid/sanctuary#125]
Please check ahead of time when declaring pipe statements if the functions are defined.
Currently stack traces when a piped function is undefined are very obscure.

Using version 0.7.1.
```js
import {pipe} from 'sanctuary';
const samplePackage = {};
const breakingFunction = pipe([samplePackage.doesNotExist]);
breakingFunction("Hello world");
```
Outputs
```
TypeError: g is not a function
    at /.../node_modules/sanctuary/index.js:257:16
    at /.../node_modules/sanctuary/index.js:395:42
    at /.../node_modules/sanctuary/index.js:246:48
    at /.../node_modules/sanctuary/index.js:179:45
    at Object.<anonymous> (/.../src/index.js:4:1)
```

**Comment by davidchambers** [comment]
Good idea, @Beanow. Fortunately, this will be resolved by #123:

```javascript
S.pipe([undefined]);
// ! TypeError: ‘pipe’ expected a value of type (List Function) as its first argument; received [undefined]
```

**Comment by davidchambers** [comment]
I've just published v0.8.0, which has the behaviour indicated in the comment above. :)
